packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx

```diff
@@ -70,13 +72,17 @@ const ActiveVisitsTable = (props) => {
   const layout = useLayoutType();
   const desktopView = layout === 'desktop';
   const config = useConfig();
+  const [currentPage, setCurrentPage] = useState(1);
   const [currentPageSize, setPageSize] = useState(config?.activeVisits?.pageSize ?? 10);
   const pageSizes = config?.activeVisits?.pageSizes ?? [10, 20, 50];
   const [loading, setLoading] = useState(true);
   const [activeVisits, setActiveVisits] = useState<ActiveVisitRow[]>([]);
   const [searchString, setSearchString] = useState('');
 
   const searchResults = useMemo(() => {
+    if (currentPage != 1) {
```

Comment:
`!==` and you do `setCurrentPage`, which is a side-effect.

Don't include side-effects in your memo computations. This is a no-go.

Reply:
Okay,  will search for a work around.

Reply:
Not a workaround. A workaround is something that helps you navigate around an obstacle. This is not an obstacle. You created a solution that is way too inefficient and complicated and now I want you to do it correctly. Simple as that.

Reply:
Sure @FlorianRappl.

Thanks alot!